File: 02-primes/primes.py
import os

target = int(os.getenv('MAXN'))

def is_prime(x):
  if x % 2 == 0:
    return False
  d = 3
  while d * d <= x:
    if x % d == 0:
      return False
    d += 2
  return True

c = 1
n = 3
while True:
  if is_prime(n):
    c += 1
    if c == target:
      print('primes[%s] == %s' % (c, n))
      break
  n += 2

File: 02-primes/primes.php
<?php

$target = getenv('MAXN');

$primes = [2, 3, 5, 7];

function is_prime($x) {
  global $primes;
  foreach ($primes as $d) {
    if ($d * $d > $x)
        return true;
    if ($x % $d == 0)
        return false;
  }
}

for ($n = 9; true; $n += 2) {
  if (is_prime($n)) {
    $primes[] = $n;
    if (count($primes) == $target) {
      echo "primes[$target] = " . $primes[$target-1] . "\n";
      break;
    }
  }
}
